dashboard/src/actions/installedpackages.ts

```diff
@@ -128,17 +130,25 @@ export function getInstalledPackage(
         );
         availablePackageDetail = resp.availablePackageDetail;
       } catch (e: any) {
-        dispatch(
-          errorInstalledPackage(
-            new FetchWarning(
-              "this package has missing information, some actions might not be available.",
+        if (e.constructor === UnauthorizedError) {
+          dispatch(actions.auth.logoutByAuthenticationError());
+        } else {
+          dispatch(
+            errorInstalledPackage(
+              new FetchWarning(
+                "this package has missing information, some actions might not be available.",
+              ),
             ),
```

Comment:
There is `actions.auth.logoutByAuthenticationError()` repeated in many places.
I was wondering if we could refactor that to be generic with something like e.g.:

```
export function handledErrorAction(err: ActionType<any>, e: any){
  if (e.constructor === UnauthorizedError) {
    return actions.auth.logoutByAuthenticationError();
  } else {
     return err;
  }
}
```

So that invocations can be almost a one-liner like e.g.:
```
dispatch(
      handledErrorAction(errorInstalledPackage(new FetchError("Unable to list apps", [e])), e)
)
```

Reply:
Good idea, thanks! Will do! 
Also, I'd like to explore the grpc interceptors at some point. Don't know if they can fit our main goal here, though. https://grpc.io/blog/grpc-web-interceptor/